client/src/utils/parser/robotCodeToSsotParsing/robotCodeToSsotParsing.js

```diff
@@ -0,0 +1,369 @@
+
+/**
+ * @category Client
+ * @module
+ */
+
+const { default: customNotification } = require("../../notificationUtils");
+
+const FOURSPACE = '    ';
+
+
+/**
+ * 
+ * @returns "uniqueId" which is just an increment from the counter in the local storage 
+ */
+const getUniqueId = () => {
+    const newId = JSON.parse(sessionStorage.getItem('idCounter')) + 1;
```

Comment:
Not really a huge fan of this, but I guess it works for now. Could we maybe use another factor (maybe date + time) into this?
Or maybe we could use [this](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues) if it is native enough and usable.

Reply:
The thing is, I'm not a fan of self-created IDs where we try to be very cryptic. Sure I can wrap time and date somehow, but parsing requires numerous IDs in milliseconds, so I see problems there. The method you suggest is great but doesn't look at whether that random string has already been returned. With my (admittedly messy) solution, we get a 100% new unique ID for each element. So I would leave it in for now (until an ID package is used).

Reply:
I was asking for another factor to be combined with this counting up, so maybe a (time+countedNumber) as Id, which should not be too hard and help us with the possibility of having conflicts.
What do you think @WolfgangDaniel